webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -5266,6 +5266,66 @@ var userOverride = {
                         $mck_box_form.addClass('mck-text-req');
                         return false;
                     }
+                    //If the field is a form field then validate the input, update user details before sending the message
+                    if (
+                        $mck_text_box.data('fieldType')
+                    ) {
+                        //If the field has a regex validation then validate the input otherwise skip validation
+                        if($mck_text_box.data('validation')){
+                            var regexForm = $mck_text_box.data(
+                                'validation'
+                            );
+                            regexForm = new RegExp(regexForm);
+                            //If the input does not match the regex validation then show an error message for 2 seconds.
+                            if (!regexForm.test(message)) {
+                                document.getElementById(
+                                    'mck-form-field-error-alert'
+                                ).innerHTML = $mck_text_box.data(
+                                    'errorMessage'
+                                );
+                                $applozic('#mck-form-field-error-alert-box')
+                                    .removeClass('n-vis')
+                                    .addClass('vis');
+                                setTimeout(function () {
+                                    $applozic('#mck-form-field-error-alert-box')
+                                        .removeClass('vis')
+                                        .addClass('n-vis');
+                                }, 2000);
+                                return false;
+                            }
+                        }
+                        if($mck_text_box.data('updateUserDetails')){
+                            // If the field is a form field and the user details need to be updated then update the user details
+                            var fieldVal = $mck_text_box.data('field');
+                            if($mck_text_box.data('fieldType') === 'EMAIL'){
```

Comment:
simplify this code, something like where the following line comes only once, replace 'X' with a simplified code
mckContactService.updateUser({
                                    data: { X: message }});